### DISCUSSION TITLE
一种update 的覆盖操作怎么写？

### DISCUSSION THREAD

**Discussion by zhza** [dotnetcore/FreeSql#1260] [Q&A] [answered]
`public class Topic {
        [Column(IsIdentity = true)]
        public int Id { get; set; }
        public int State { get; set; }
        public string Name { get; set; }

        [Navigate(nameof(Snapshot.Tid))]
        public Snapshot Related { get; set; }

        //..... other fields
    }

    public class Snapshot
    { 
        public int Tid { get; set; }
        public int State { get; set; }
    }`

需要用  Snapshot的 State 字段替换 Topic表的State字段

var sql = repo.Select
                .Include(a => a.Related)
                .ToUpdate()
                .Set(a => a.State == a.Related.State)              
                .ToSql();
这样写不太对，直接报找不到Related字段。
请教正确的应该如何写？

**Comment by 2881099** [comment, marked as answer]
ToUpdate 功能有限，只能联表查询，不支持联表赋值更新

**Comment by 2881099** [comment]
联表更新，属于危险操作，FreeSql 为了使用者安全不提供太简单联表的更新操作
